### PR TITLE
Renamed cost allocation tags to cost tags and added schema for get list of all available tag keys

### PR DESCRIPTION
**Pull request by asarkar84**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by asarkar84** [comment]
@jianghaolu Please take a look. Thanks!

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-03-31/consumption.json [hunk 1]
```diff
@@ -1868,6 +1907,43 @@
         }
       }
     },
+    "Tags": {
+      "description": "A tag resource.",
```
Comment (jianghaolu) [inline specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-03-31/consumption.json]: Please put in more detailed descriptions. This is easily confused with "Tag".

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-03-31/examples/Tags.json [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "id": "providers/Microsoft.CostManagement/billingAccounts/{billingaccount-id}/providers/Microsoft.Consumption/tags/tags1",
+        "name": "tags1",
+        "type": "Microsoft.Consumption/tags",
+        "eTag": "\"1d34d012214157f\"",
+        "properties": {
+          "tags": [
```
Comment (jianghaolu) [inline specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-03-31/examples/Tags.json]: The property name here should be "costTags", as defined in `TagProperties`.

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-03-31/consumption.json (hunk 1 above)
Comment (jianghaolu) [reply specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-03-31/consumption.json]: You might even consider renaming this to "TagResource".

**Review by jianghaolu — CHANGES_REQUESTED** [review]

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-03-31/examples/Tags.json (hunk 2 above)
Comment (asarkar84) [reply specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-03-31/examples/Tags.json]: Thank you! for mentioning this. Actually it was wrong at the other place. I am corrected it there.

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-03-31/consumption.json (hunk 1 above)
Comment (asarkar84) [reply specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-03-31/consumption.json]: I have updated the description. Please take a look if the changes are fine.

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-03-31/consumption.json [hunk 3]
```diff
@@ -1868,6 +1907,43 @@
...
+          "$ref": "#/definitions/TagProperties",
+          "title": "Tag properties"
+        }
+      }
+    },
+    "TagProperties": {
+      "description": "The properties of the tag.",
+      "properties": {
+        "tags": {
+          "description": "Tags.",
```
Comment (jianghaolu) [inline specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-03-31/consumption.json]: One more thing : Please avoid using property name as the description. Try to add a little more info :)

Comment (asarkar84) [reply specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-03-31/consumption.json]: Sure! Will keep this in mind for future as well.

**Comment by jianghaolu** [comment]
Per offline discussion - this is a breaking change but 4/13/18 is when the service is first deployed. So this is accepted for now and no future breaking changes will be allowed to this spec.

**Review by jianghaolu — APPROVED** [review]

**Pull request merged**